### DISCUSSION TITLE
Generating jax-rs resources with panache: Total records?

### DISCUSSION THREAD

**Discussion by NMichas** [quarkusio/quarkus#26316] [Q&A] [answered]
Hello,

Quarkus provides a convenient functionality to quickly expose panache-backed data via jax-rs as described here: https://quarkus.io/guides/rest-data-panache.

However, I can't find any configuration option for my paginated results to contain the "number of total records". Is this somehow supported?

Thanks!

[a comment by an automated account is not shown]

**Comment by FroMage** [comment]
Good point. There's https://quarkus.io/guides/rest-data-panache#get-peoplepage0size2 which shows that you can use the `last` link to know which page is the last one, but it doesn't tell you how many records there are. I don't think we can include it in the `list` result, since it's an array. Unless we add it as a response header. It's also not a link resource. Perhaps it should be a new URI such as `GET /people/size`?

Any opinion @geoand ?

**Reply by geoand** [reply]
Sounds reasonable to me.

cc @Sgitario

**Reply by Sgitario** [reply]
+1 will work in this: adding a new endpoint for panache / panache reactive to get the total size of records.

**Reply by Sgitario** [reply, marked as answer]
https://github.com/quarkusio/quarkus/pull/27193

**Comment by codespearhead** [comment]
@NMichas The "/count" endpoint was added in #27193 , thanks to @Sgitario . Can you mark this discussion as answered?
